src/NuGet.Core/NuGet.Packaging/Licenses/LicenseExpressionParser.cs

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace NuGet.Packaging
+{
+    public class LicenseExpressionParser
+    {
+
+        /// <summary>
+        /// Parses a License Expression if valid.
+        /// The expression would be parsed correct, even if non-standard exceptions are encountered. The non-standard Licenses/Exceptions have metadata on them with which the caller can make decisions.
+        /// Based on the Shunting Yard algorithm. <see href="https://en.wikipedia.org/wiki/Shunting-yard_algorithm"/>
+        /// This method first creates an postfix expression by separating the operators and operands.
+        /// Later the postfix expression is evaluated into an object model that represents the expression. Note that brackets are dropped in this conversion and this is not round-trippable.
+        /// The token precedence helps make sure that the expression is a valid infix one. 
+        /// </summary>
+        /// <param name="infixTokens"></param>
```

Comment:
Remove `params` tag for this parameter that doesn't exist and add one for `expression`